src/solvers/dgmulti/flux_differencing_gauss_sbp.jl

```diff
@@ -25,15 +25,262 @@ function tensor_product_quadrature(element_type::Hex, r1D, w1D)
+                                        dg::DGMulti{3, Hex, GaussSBP})
+  rd = dg.basis
+
+  rq1D, wq1D = StartUpDG.gauss_quad(0, 0, polydeg(dg))
+  interp_matrix_gauss_to_face_1d = polynomial_interpolation_matrix(rq1D, [-1; 1])
+
+  nnodes_1d = length(rq1D)
+
+  # Permutation of indices in a tensor product form
+  indices = reshape(1:length(rd.rf), nnodes_1d, nnodes_1d, rd.Nfaces)
+  face_indices_tensor_product = zeros(Int, 2, nnodes_1d, nnodes_1d, ndims(rd.elementType))
+  for i in 1:nnodes_1d, j in 1:nnodes_1d # loop over nodes in one face
+    face_indices_tensor_product[:, i, j, 1] .= indices[i, j, 1:2]
+    face_indices_tensor_product[:, i, j, 2] .= indices[i, j, 3:4]
+    face_indices_tensor_product[:, i, j, 3] .= indices[i, j, 5:6]
+  end
+
+  T_op = typeof(operator)
+  Tm = typeof(interp_matrix_gauss_to_face_1d)
+  Tw = typeof(rd.wq)
+  Ti = typeof(face_indices_tensor_product)
+  return TensorProductGaussFaceOperator{3, T_op, Tm, Tw, Ti}(interp_matrix_gauss_to_face_1d,
+                                                             inv.(wq1D), face_indices_tensor_product,
+                                                             nnodes_1d, rd.Nfaces)
+end
+
+@inline function mul_by!(A::TensorProductGaussFaceOperator)
+  return (out, x) -> tensor_product_gauss_face_operator!(out, A, x)
+end
 
+@inline function tensor_product_gauss_face_operator!(out::AbstractMatrix,
+                                                     A::TensorProductGaussFaceOperator,
+                                                     x::AbstractMatrix)
+  @threaded for col in Base.OneTo(size(out, 2))
+    tensor_product_gauss_face_operator!(view(out, :, col), A, view(x, :, col))
+  end
+end
 
 # By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
 # Since these FMAs can increase the performance of many numerical algorithms,
 # we need to opt-in explicitly.
 # See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
 @muladd begin
 
+# Interpolates values from volume Gauss nodes to face nodes on one element.
+@inline function tensor_product_gauss_face_operator!(out::AbstractVector,
+                                                     A::TensorProductGaussFaceOperator{2, Interpolation},
+                                                     x::AbstractVector)
+
+  @unpack interp_matrix_gauss_to_face_1d, face_indices_tensor_product = A
+  @unpack nnodes_1d, nfaces = A
+
+  fill!(out, zero(eltype(out)))
+  x = reshape(x, nnodes_1d, nnodes_1d)
+
+  # interpolation in the x-direction
+  @turbo for i in Base.OneTo(nnodes_1d) # loop over nodes in a face
+    index_left = face_indices_tensor_product[1, i, 1]
+    index_right = face_indices_tensor_product[2, i, 1]
+    for jj in Base.OneTo(nnodes_1d)      # loop over "line" of volume nodes
+      out[index_left] = out[index_left] + interp_matrix_gauss_to_face_1d[1, jj] * x[jj, i]
+      out[index_right] = out[index_right] + interp_matrix_gauss_to_face_1d[2, jj] * x[jj, i]
```

Comment:
```suggestion
    index_left  = face_indices_tensor_product[1, i, 1]
    index_right = face_indices_tensor_product[2, i, 1]
    for jj in Base.OneTo(nnodes_1d)      # loop over "line" of volume nodes
      out[index_left]  = out[index_left]  + interp_matrix_gauss_to_face_1d[1, jj] * x[jj, i]
      out[index_right] = out[index_right] + interp_matrix_gauss_to_face_1d[2, jj] * x[jj, i]
```
This makes it easier for me to spot the differences and see that it looks fine.